theano/gpuarray/tests/test_elemwise.py

```diff
@@ -40,16 +40,21 @@ def test_elemwise_pow():
         for dtype_exp in dtypes:
 
             # Compile a gpu function with the specified dtypes
-            base = theano.tensor.vector(dtype=dtype_base)
-            exp = theano.tensor.vector(dtype=dtype_exp)
-            output = base ** exp
-            f = theano.function([base, exp], output)
-
             base_val = np.random.randint(0, 5, size=10).astype(dtype_base)
             exp_val = np.random.randint(0, 3, size=10).astype(dtype_exp)
 
+            base = theano.tensor.vector(dtype=dtype_base)
+            exp = gpuarray_shared_constructor(exp_val)
```

Comment:
I would add a check here to make sure that `exp.dtype == dtype_exp`, to make sure we are actually testing the right thing.